### ISSUE TITLE
LowLevel Support for using PUT to set Settings and Mappings at same time

### ISSUE THREAD

**Issue by Airn5475** [elastic/elasticsearch-net#2518]
I am trying to set up an Edge NGram Tokenizer on my index.  I currently create the index mapping using `Client.LowLevel.IndicesPutMappingPost` which I get from a .json file.

I was hoping to add Settings to this .json file and submit them along with the mapping in one shot.

Example from [Documentation](https://www.elastic.co/guide/en/elasticsearch/reference/5.1/analysis-edgengram-tokenizer.html):

```
PUT my_index
{
  "settings": {
    "analysis": {
      "analyzer": {
        "autocomplete": {
          "tokenizer": "autocomplete",
          "filter": [
            "lowercase"
          ]
        },
        "autocomplete_search": {
          "tokenizer": "lowercase"
        }
      },
      "tokenizer": {
        "autocomplete": {
          "type": "edge_ngram",
          "min_gram": 2,
          "max_gram": 10,
          "token_chars": [
            "letter"
          ]
        }
      }
    }
  },
  "mappings": {
    "doc": {
      "properties": {
        "title": {
          "type": "text",
          "analyzer": "autocomplete",
          "search_analyzer": "autocomplete_search"
        }
      }
    }
  }
}
```

I tried using `Client.LowLevel.IndexPut` but this requires a `type`, which is not allowed in the example above.

**Comment by Mpdreamz** [comment]
PUT my_index


Is exposed under `client.LowLevel.IndicesCreate` 

`IndexPut` is a PUT operation on `/index/type/id` e.g the Index API

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#_automatic_id_generation

**Comment by tmenier** [comment]
Ahhh, glad I found this. Sorry to offer negative feedback, but the naming isn't very intuitive, especially for a method that maps to [the most basic/common way of creating an index](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html). I was pulling out what's left of my hair trying to understand why `IndexPut` required a type. ;)

Obviously it would be a breaking change but naming conventions might be something to re-address in a future major release.

**Comment by Mpdreamz** [comment]
@tmenier thanks for the feedback, although we want to keep the breaking changes to a minimum I also would love to go over the naming of methods in the upcoming 7.0 release.

Most folks don't use the low-level client directly so it might be okay to do so.

**Comment by tmenier** [comment]
@Mpdreamz Thanks, glad that was well taken. I'm probably a minority case in that I actually moved off of NEST after using it for a while. It's a really slick SDK, but I found the learning curve was just high enough (or my memory of its conventions just poor enough) that I found everything I wanted to do I had to learn twice: once how to do it in Elastic and once how to do it in NEST. Now my code is a little fatter, but once I learn how to do something in Elastic everything else I need is right in front of me. Suits my style a little better. Again, I'm probably in the minority, but I am one use case for the low-level client.
